### PR TITLE
docs: added docs for migrating encrypted secrets

### PR DESCRIPTION
**Pull request by justo-mend**
## Changes

Updated docs to show how to migrate encrypted secrets.

## Context

Storing encrypted secrets in Renovate config is deprecated and will no longer be supported from 01-Sep-2024. New docs have been added to show how to migrate encrypted secrets from Renovate config to the App settings via the web UI.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [X] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/usage/getting-started/app-secrets.md
Comment (rarkins) [inline docs/usage/getting-started/app-secrets.md:1]: Use one sentence per line, avoid html use

File: docs/usage/getting-started/app-secrets.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# Using Secrets with Mend Cloud Apps
+
+The Mend-hosted cloud apps - Renovate App on GitHub, Mend App on Bitbucket - have deprecated the use of encrypted secrets in Renovate config files stored in the repo.
+Instead, secrets for the cloud apps must be stored in the app settings using the web UI available at [developer.mend.io](http://developer.mend.io).
+
+## How to use secrets in the Mend Hosted Cloud Apps
+
+To use secrets in the Mend Cloud apps:
+
+1. Add the secret in the **Credentials** section of the [Developer Portal](http://developer.mend.io) Org or Repo settings.
```
Comment (rarkins) [inline docs/usage/getting-started/app-secrets.md]: Need to mention here that only users with admin rights can add secrets

File: docs/usage/getting-started/migrating-secrets.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
+# Migrating Secrets from Repo Config to App Settings
+
+Storing of encrypted secrets in the repo is not supported starting from 01-Sep-2024 for Mend Renovate Cloud apps.
```
Comment (rarkins) [inline docs/usage/getting-started/migrating-secrets.md:3]: I'm not sure this date is realistic if we haven't yet begun to warn users

**Review by rarkins — CHANGES_REQUESTED** [review]

File: docs/usage/getting-started/migrating-secrets.md (hunk 2 above)
Comment (justo-mend) [reply docs/usage/getting-started/migrating-secrets.md:3]: I'll bump it to 1-Oct-2024

File: docs/usage/getting-started/app-secrets.md
Comment (justo-mend) [reply docs/usage/getting-started/app-secrets.md:1]: Updated. Removed BR tags from app-secrets.md

File: docs/usage/getting-started/app-secrets.md (hunk 1 above)
Comment (justo-mend) [reply docs/usage/getting-started/app-secrets.md]: Updated. Added note.

File: docs/usage/assets/images/app-settings/repo-settings-button.png
Comment (HonkingGoose) [inline docs/usage/assets/images/app-settings/repo-settings-button.png:1]: It's hard to see if you're on the Organization page, or on the Repository page. Each page should be clearly different. Making these changes will help:

| Screen for | Logo | Button text | Heading text |
| --- | --- | --- | --- |
| Organization | Organization avatar | Organization settings | `$organization` overview |
| Repository | Generic repository logo | Repository settings | `$repository` overview |

File: docs/usage/getting-started/migrating-secrets.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+3. In the **Credentials** page, press `ADD SECRET` to add the encrypted secret.
+
+4. Provide a value for `Secret name`, put the encrypted secret in the `Secret Value` field, and press `SAVE`.
+
+   > [!NOTE]
+   >
+   > When migrating an existing secret from a repo, be sure to add the secret to the same Org or Repo for which the secret was generated.
+   >
+   > - A secret generated for a specific repo cannot be added to a different repo.
```
Comment (HonkingGoose) [inline docs/usage/getting-started/migrating-secrets.md:55]:
```suggestion
   > - A secret generated for a specific repository can only be added to that _same_ repository.
```

File: docs/usage/getting-started/migrating-secrets.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+   >   - A secret generated for a specific repo cannot be added to the Org settings. It can only be added into Repo settings for the matching repo.
```
Comment (HonkingGoose) [inline docs/usage/getting-started/migrating-secrets.md:56]:
```suggestion
   >   - A secret generated for a specific repository can only be added to the repository settings for the matching repository. This secret can _not_ be added to the organization's settings.
```

File: docs/usage/getting-started/migrating-secrets.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+   >   - A secret generated without a specific repo can be added into Org or Repo settings in the matching org.
```
Comment (HonkingGoose) [inline docs/usage/getting-started/migrating-secrets.md:57]:
```suggestion
   >   - A secret generated without a specific repository _can_ be added into the organization _or_ into the repository settings under that organization.
```

File: docs/usage/getting-started/migrating-secrets.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+   The settings UI will provide helpful information if you try to add the secret in the wrong place.
```
Comment (HonkingGoose) [inline docs/usage/getting-started/migrating-secrets.md:59]:
```suggestion
   When you try to add a secret in the wrong place, the settings UI will help you:
```

File: docs/usage/getting-started/migrating-secrets.md [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+
+   ![Migrating secrets error](../assets/images/app-settings/encrypted-secrets-error.png)
+
+5. Look for the confirmation box to say **“Successfully migrated secret”** to confirm that the secret was correctly decrypted and stored in the correct Org or Repo.
```
Comment (HonkingGoose) [inline docs/usage/getting-started/migrating-secrets.md:63]:
```suggestion
5. When you see the confirmation box with the text: **“Successfully migrated secret”**, your secret is now stored in the correct organization or repository.
```

Do we need the detail about the decrypting? If yes, we should probably mention if/that you re-encrypted the value safely... :wink:

File: docs/usage/getting-started/migrating-secrets.md [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+
+![Successfully migrated secret](../assets/images/app-settings/stored-secret-encrypted.png)
+
+> [!IMPORTANT]
+>
+> If the confirmation box does not say **“Successfully migrated secret”**, and instead says **“Successfully stored secret”** then the secret was not detected as a Renovate encrypted secret and the value was treated as plain text.
```
Comment (HonkingGoose) [inline docs/usage/getting-started/migrating-secrets.md:69]:
```suggestion
> If the confirmation box does _not_ say **“Successfully migrated secret”**, but says: **“Successfully stored secret”** then the secret was not detected as a Renovate encrypted secret, and the value was treated as plaintext.
```

File: docs/usage/getting-started/migrating-secrets.md [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+> If you were expecting to import a secret originally encrypted by Renovate:
+>
+> - Make sure you are using the secret in the correct Org and/or Repo.
```
Comment (HonkingGoose) [inline docs/usage/getting-started/migrating-secrets.md:72]:
```suggestion
> - Make sure you are pasting the secret into the correct organization or repository
```

File: docs/usage/getting-started/migrating-secrets.md [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+> - Check that you copied the encrypted secret correctly, and try again.
+
+## To migrate encrypted secrets using the plain text value
+
+1. Go to the settings for the correct Org or Repo in the web UI at [http://developer.mend.io](http://developer.mend.io).
```
Comment (HonkingGoose) [inline docs/usage/getting-started/migrating-secrets.md:77]:
```suggestion
1. Go to the settings for the correct organization or repository in the web UI at [developer.mend.io](https://developer.mend.io).
```

File: docs/usage/getting-started/migrating-secrets.md [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+
+2. In the **Credentials** page, press `ADD SECRET` to add the plain text secret.
```
Comment (HonkingGoose) [inline docs/usage/getting-started/migrating-secrets.md:79]:
```suggestion
2. On the **Credentials** page, select `ADD SECRET` to add the plaintext secret.
```

File: docs/usage/getting-started/migrating-secrets.md [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+
+   ![Add repo secret](../assets/images/app-settings/add-repo-secret.png)
+
+3. Provide a value for `Secret name`, put the secret in the `Secret Value` field, and press `SAVE`.
```
Comment (HonkingGoose) [inline docs/usage/getting-started/migrating-secrets.md:83]:
```suggestion
3. Give a value for `Secret name`, paste the plaintext secret into the `Secret Value` field, and select `SAVE`.
```

File: docs/usage/getting-started/migrating-secrets.md [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+
+   ![Add a Secret dialog box](../assets/images/app-settings/add-a-secret.png)
+
+4. Look for a confirmation dialog showing **“Successfully stored secret”**.
```
Comment (HonkingGoose) [inline docs/usage/getting-started/migrating-secrets.md:87]:
```suggestion
4. Wait for the confirmation dialog: **“Successfully stored secret”**.
```

File: docs/usage/getting-started/private-packages.md [hunk 14]
```diff
@@ -585,30 +585,32 @@ The solution to this is that you should break your presets into public and priva
 It is strongly recommended that you avoid committing secrets to repositories, including private ones, and this includes secrets needed by Renovate to access private modules.
 The preferred approach to secrets is that the bot administrator configures them as `hostRules` which are then applied to all repositories which the bot accesses.
 
+> [!IMPORTANT]
+>
+> Use of encrypted secrets in Renovate config has been deprecated in the Mend Renovate App. Secrets must now be stored in the App settings via the web UI.
+>
+> For more information, see [Migrating Secrets from Repo Config to App Settings](migrating-secrets.md)
```
Comment (HonkingGoose) [inline docs/usage/getting-started/private-packages.md:592]:
```suggestion
<!-- prettier-ignore -->
!!! warning "Store secrets for your Mend-hosted app via the web UI"
    Mend no longer supports putting encrypted secrets in the Renovate config file on your repository.
    Going forward, all secrets must be stored in the App settings via the web UI.
    If you have encrypted secrets in your Renovate config, you must migrate them to the web UI.
    Read [Migrating Secrets from Repo Config to App Settings](migrating-secrets.md) to learn how.
```

File: docs/usage/getting-started/private-packages.md [hunk 15]
```diff
@@ -585,30 +585,32 @@ The solution to this is that you should break your presets into public and priva
...
-      "matchHost": "https://custom.registry.company.com/pypi/",
-      "username": "bot1",
-      "encrypted": {
-        "password": "p278djfdsi9832jnfdshufwji2r389fdskj........."
-      }
-    }
-  ]
-}
-```
+   ![Organisation and Repository secrets in Credentials settings page](../assets/images/app-settings/org-and-repo-secrets.png)
```
Comment (HonkingGoose) [inline docs/usage/getting-started/private-packages.md:598]:
```suggestion
   ![Organization and repository secrets on the credentials settings page](../assets/images/app-settings/org-and-repo-secrets.png)
```

**Review by HonkingGoose — CHANGES_REQUESTED** [review]
I changed a whole lot! So please double check that the information is technically correct.

The docs has text about the confirmation box at the final step of importing a secret. It basically says that the reader should check the message carefully, to see if their secret is in "state A" or "state B". That kind of information _must_ go in the confirmation box itself, in _clear_ words.

At the end, as user you want to know:

- Can Renovate see/use my secret now?
- Is my secret still safe?
- Is this what I expected to happen? If not, how can I get what I want?

The web UI should guide and help the user through the process. The docs are supporting material.

File: docs/usage/assets/images/app-settings/repo-settings-button.png
Comment (rarkins) [reply docs/usage/assets/images/app-settings/repo-settings-button.png:1]: Not a part of this PR though, let's discuss separately

**Review by rarkins — DISMISSED** [review]

File: docs/usage/assets/images/app-settings/repo-settings-button.png
Comment (HonkingGoose) [reply docs/usage/assets/images/app-settings/repo-settings-button.png:1]: See:

- #30273

File: docs/usage/getting-started/migrating-secrets.md [hunk 16]
```diff
@@ -0,0 +1,93 @@
+# Migrating Secrets from Repo Config to App Settings
+
+On 01-Oct-2024 the Mend Renovate cloud apps will stop reading any encrypted secrets from the Renovate configuration file on your repository.
+Previously, you could encrypt a secret with the [Renovate encryption tool](https://app.renovatebot.com/encrypt) and then put it in your Renovate config file.
+Going forward, all secrets must be stored in the App settings on the cloud.
+They can be referenced from the Renovate config files inside the repo using `{{ secrets.SECRET_NAME }}` notation.
+
+
```
Comment (HonkingGoose) [inline docs/usage/getting-started/migrating-secrets.md:8]:
```suggestion
```

File: docs/usage/getting-started/migrating-secrets.md [hunk 17]
```diff
@@ -0,0 +1,93 @@
...
+      "matchHost": "github.com",
+      "encrypted": {
+        "token": "drsMDVf6M2hTZCN......+gQm/0Rpw"
+      }
+    }
+  ]
+}
+```
+
+
```
Comment (HonkingGoose) [inline docs/usage/getting-started/migrating-secrets.md:22]:
```suggestion
```

**Review by HonkingGoose — CHANGES_REQUESTED** [review]
Fixes for the lint errors.

**Review by rarkins — DISMISSED** [review]

**Comment by rarkins** [comment]
@justo-mend in future please don't submit PRs from your `main` branch

File: docs/usage/getting-started/app-secrets.md [hunk 18]
```diff
@@ -0,0 +1,83 @@
...
+
+![organization settings button](../assets/images/app-settings/org-settings-button.png)
+
+### Managing repository-level secrets
+
+The **Recent jobs** table means you are on your repository's page.
+Select the _Settings_ button to manage your repository secrets:
+
+![repository settings button](../assets/images/app-settings/repo-settings-button.png)
+
```
Comment (rarkins) [inline docs/usage/getting-started/app-secrets.md:83]:
```suggestion
```

**Review by rarkins — APPROVED** [review]

**Comment by jamietanna** [comment]
FYI this has introduced a linting error into the docs on HEAD, just spotted it on my own PR!

**Pull request merged**